src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkBaseExternalContext.java

```diff
@@ -223,6 +223,46 @@ private static ImmutableMap<URI, Map<String, List<String>>> getAuthHeaders(
     return res;
   }
 
+  private static ImmutableMap<URI, Map<String, List<String>>> getHeaders(
+    Map<String, Map<String, List<String>>> auth) throws RepositoryFunctionException, EvalException {
+    ImmutableMap.Builder<URI, Map<String, List<String>>> headers = new ImmutableMap.Builder<>();
+    for (Map.Entry<String, Map<String, List<String>>> entry : auth.entrySet()) {
+      try {
+        URL url = new URL(entry.getKey());
+        Map<String, List<String>> headerMap = entry.getValue();
+        headers.put(url.toURI(), headerMap);
+      } catch (MalformedURLException e) {
+        throw new RepositoryFunctionException(e, Transience.PERSISTENT);
+      } catch (URISyntaxException e) {
+        throw new EvalException(e);
+      }
+    }
```

Comment:
This is fine functionally, just a few suggestions on code style. Similar changes can be applied below as well.

```suggestion
    ImmutableMap.Builder<URI, Map<String, List<String>>> headers =
        ImmutableMap.builderWithExpectedSize(auth.size());
    for (var entry : auth.entrySet()) {
      try {
        headers.put(new URL(entry.getKey()).toURI(), entry.getValue());
      } catch (MalformedURLException e) {
        throw new RepositoryFunctionException(e, Transience.PERSISTENT);
      } catch (URISyntaxException e) {
        throw new EvalException(e);
      }
    }
```